### PR TITLE
BAEL- 3482

### PR DESCRIPTION
**Pull request by tinomthomas**
Initial commit

### CODE REVIEW

File: algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            heapify(lastLeafsParentIndex);
+            lastLeafsParentIndex--;
+        }
+    }
+
+    void heapify(int index) {
+        int leftNodeIndex = getLeftNodeIndex(index);
+        int rightNodeIndex = getRightNodeIndex(index);
+        int smallestElementIndex = index;
+        if (leftNodeIndex < heapNodes.length && heapNodes[leftNodeIndex].element < heapNodes[index].element)
```
Comment (eric-martin) [inline algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java]: Use { } even for one line if/else bodies.

File: algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            smallestElementIndex = leftNodeIndex;
+
+        if (rightNodeIndex < heapNodes.length && heapNodes[rightNodeIndex].element < heapNodes[smallestElementIndex].element)
```
Comment (eric-martin) [inline algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java]: Use { } here also

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.baeldung.algorithms.minheapmerge;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+public class MinHeapmergeUnitTest {
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: rename to match the case of the class under test: MinHeapMergeUnitTest

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import org.junit.Test;
+
+public class MinHeapmergeUnitTest {
+
+    private final int[][] inputArray = { { 0, 6 }, { 1, 5, 10, 100 }, { 2, 4, 200, 650 } };
+    private final int[] expectedArray = { 0, 1, 2, 4, 5, 6, 10, 100, 200, 650 };
+
+    @Test
+    public void givenSortedArraysWhenMergedShouldReturnASingleSortedarray() {
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]:
```suggestion
    public void givenSortedArrays_whenMerged_thenShouldReturnASingleSortedarray() {
```

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        int[] resultArray = MinHeapMerge.merge(inputArray);
+        assertThat(resultArray.length, is(equalTo(10)));
+        assertThat(resultArray, is(equalTo(expectedArray)));
+    }
+    
+    @Test
+    public void givenEmptyHeapNodesAndInputArrayWhenPopulatedShouldPopulateAndReturnResultArrayLength() {
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]:
```suggestion
    public void givenEmptyHeapNodesAndInputArray_whenPopulated_thenShouldPopulateAndReturnResultArrayLength() {
```

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        int size = MinHeapMerge.populateHeapNodesAndDetermineResultingArrayLength(inputArray, new HeapNode[inputArray.length]);
+        assertThat(size, is(equalTo(10)));
+    }
+    
+    @Test
+    public void givenArrayAndHeapNodesAndResultArraySizeWhenMergedShouldReturnASingleSortedarray() {
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]:
```suggestion
    public void givenArrayAndHeapNodesAndResultArraySize_whenMerged_thenShouldReturnASingleSortedarray() {
```

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+public class MinHeapmergeUnitTest {
+
+    private final int[][] inputArray = { { 0, 6 }, { 1, 5, 10, 100 }, { 2, 4, 200, 650 } };
+    private final int[] expectedArray = { 0, 1, 2, 4, 5, 6, 10, 100, 200, 650 };
+
+    @Test
+    public void givenSortedArraysWhenMergedShouldReturnASingleSortedarray() {
+        int[] resultArray = MinHeapMerge.merge(inputArray);
+        assertThat(resultArray.length, is(equalTo(10)));
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: insert a blank line before the assertion

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        assertThat(resultArray, is(equalTo(expectedArray)));
+    }
+    
+    @Test
+    public void givenEmptyHeapNodesAndInputArrayWhenPopulatedShouldPopulateAndReturnResultArrayLength() {
+        int size = MinHeapMerge.populateHeapNodesAndDetermineResultingArrayLength(inputArray, new HeapNode[inputArray.length]);
+        assertThat(size, is(equalTo(10)));
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: blank line here too

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    
+    @Test
+    public void givenArrayAndHeapNodesAndResultArraySizeWhenMergedShouldReturnASingleSortedarray() {
+        HeapNode[] heapNodes = new HeapNode[inputArray.length];
+        int size = MinHeapMerge.populateHeapNodesAndDetermineResultingArrayLength(inputArray, heapNodes);
+        int[] resultArray = MinHeapMerge.createMinHeapAndMergeArrays(inputArray, heapNodes, size);
```
Comment (KevinGilmore) [inline algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: blank line before and after this line (separates the given, when, and then portions)

File: algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java (hunk 1 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java]: Done

File: algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java (hunk 2 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/main/java/com/baeldung/algorithms/minheapmerge/MinHeap.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 3 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 4 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 6 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 5 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 7 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 9 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

File: algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java (hunk 8 above)
Comment (tinomthomas) [reply algorithms-miscellaneous-5/src/test/java/com/baeldung/algorithms/minheapmerge/MinHeapmergeUnitTest.java]: Done

**Pull request merged**